src/Adapter/Category/QueryHandler/GetCategoryForEditingHandler.php

```diff
@@ -0,0 +1,197 @@
+            $category->id_parent,
+            $category->meta_title,
+            $category->meta_description,
+            $category->meta_keywords,
+            $category->link_rewrite,
+            $category->getGroups(),
+            $category->getAssociatedShops(),
+            (bool) $category->is_root_category,
+            $this->getCoverImage($query->getCategoryId()),
+            $this->getThumbnailImage($query->getCategoryId()),
+            $this->getMenuThumbnailImages($query->getCategoryId())
+        );
+
+        return $editableCategory;
+    }
+
+    /**
+     * @param CategoryId $categoryId
+     *
+     * @return array
+     */
+    private function getCoverImage(CategoryId $categoryId)
+    {
+        $imageType = 'jpg';
+        $image = _PS_CAT_IMG_DIR_ . $categoryId->getValue() . '.' . $imageType;
+
+        $imageTag = ImageManager::thumbnail(
+            $image,
+            'category' . '_' . $categoryId->getValue() . '.' . $imageType,
+            350,
+            $imageType,
+            true,
+            true
+        );
+
+        $imageSize = file_exists($image) ? filesize($image) / 1000 : '';
+
+        if (empty($imageTag) || empty($imageSize)) {
+            return null;
+        }
+
+        return [
+            'size' => sprintf('%skb', $imageSize),
+            'path' => $this->imageTagSourceParser->parse($imageTag),
+        ];
+    }
+
+    /**
+     * @param CategoryId $categoryId
+     *
+     * @return array
+     */
+    private function getThumbnailImage(CategoryId $categoryId)
+    {
+        $image = _PS_CAT_IMG_DIR_ . $categoryId->getValue() . '.jpg';
+        $imageTypes = ImageType::getImagesTypes('categories');
+
+        $thumb = '';
+        $imageTag = '';
+        $formatted_small = ImageType::getFormattedName('small');
+        foreach ($imageTypes as $k => $image_type) {
+            if ($formatted_small == $image_type['name']) {
+                $thumb = _PS_CAT_IMG_DIR_ . $categoryId->getValue() . '-' . $image_type['name'] . '.jpg';
+                if (is_file($thumb)) {
+                    $imageTag = ImageManager::thumbnail(
+                        $thumb,
+                        'category_' . (int) $categoryId->getValue() . '-thumb.jpg',
+                        (int) $image_type['width'],
+                        'jpg',
+                        true,
+                        true
+                    );
+                }
+            }
+        }
+
+        if (!is_file($thumb)) {
+            $thumb = $image;
+            $imageTag = ImageManager::thumbnail($image, 'category_' . $categoryId->getValue() . '-thumb.jpg', 125, 'jpg', true, true);
+            ImageManager::resize(_PS_TMP_IMG_DIR_ . 'category_' . $categoryId->getValue() . '-thumb.jpg', _PS_TMP_IMG_DIR_ . 'category_' . $categoryId->getValue() . '-thumb.jpg', (int) $image_type['width'], (int) $image_type['height']);
+        }
+
+        $thumbSize = file_exists($thumb) ? filesize($thumb) / 1000 : false;
```

Comment:
should be 1024 with filesize no?

Reply:
nope, 1024 would be for `KiB` and 1000 is for `kB`.

https://i.imgur.com/faKgB6e.png

Reply:
@jolelievre so true, that's why I prefer the usage of kiB instead of kB  ^^